### PR TITLE
Update cutadapt v.3.4

### PR DESCRIPTION
**Pull request by gallardoalba**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

This PR updates cutadapt to version 3.4. It includes the following modifications:
- Include option `--cut`: [info](https://cutadapt.readthedocs.io/en/stable/guide.html#removing-a-fixed-number-of-bases)
- Include option `--rename`: [info](https://cutadapt.readthedocs.io/en/stable/guide.html#rename-renames-reads)
- Include option `--action`: [info](https://cutadapt.readthedocs.io/en/stable/guide.html#action-changes-what-is-done-when-an-adapter-is-found)
- Include option `--revcomp`: [info](https://cutadapt.readthedocs.io/en/stable/guide.html#searching-reverse-complements)
- Include the possibility of setting minimum and maximum length in R2 in paired reads: [info](https://cutadapt.readthedocs.io/en/stable/guide.html#filtering-paired-end-reads)
- Remove the options `--prefix` and `--suffix`, since are considered outdated: [info](https://cutadapt.readthedocs.io/en/stable/guide.html#modifying-read-names)
- Update the `--pair-filter`option (include _first_): [info](https://cutadapt.readthedocs.io/en/stable/guide.html#filtering-paired-end-reads)
- Include the `Disallow internal adaptor ocurrences` option: [info](https://cutadapt.readthedocs.io/en/stable/guide.html#non-internal-5-and-3-adapters)
- Include the option `--max-expected-errors`: [info](https://cutadapt.readthedocs.io/en/stable/guide.html#filtering-reads)
- Reorganize output options section.

### CODE REVIEW

**Comment by gallardoalba** [comment]
#2585

File: tools/cutadapt/cutadapt.xml [hunk 1]
```diff
@@ -527,19 +557,127 @@ $read_mod_options.trim_n
...
+                </conditional>
+            </section>
+            <output name="out1" decompress="True" file="cutadapt_out1_min_length.fq.gz" ftype="fastq.gz"/>
+            <output name="out2" decompress="True" file="cutadapt_out2_min_length.fq.gz" ftype="fastq.gz"/>
+            <assert_command>
+                <has_text text="--minimum-length=30:10"/>
+                <has_text text="--pair-filter=both"/>
+            </assert_command>
+        </test>
+                <!-- Test maximum lenghth paired-reads -->
```
Comment (bgruening) [inline tools/cutadapt/cutadapt.xml:657]: indentation

**Review by bgruening — APPROVED** [review]

File: tools/cutadapt/cutadapt.xml [hunk 2]
```diff
@@ -187,128 +193,157 @@ $read_mod_options.trim_n
...
+            <param argument="--trim-n" type="boolean" truevalue="--trim-n" falsevalue="" checked="False" label="Trim Ns" help="Trim N's on ends of reads." />
+            <param argument="--strip-suffix" label="Strip suffix" type="text" help="Remove this suffix from read names if present." />
+            <param argument="--length" type="integer" value="0" label="Length" help="Shorten reads to this length. This modification is applied after adapter trimming." />
+            <param argument="--length-tag" label="Length Tag" type="text" help="Search for TAG followed by a decimal number in the name of the read (description/comment field of the FASTA or FASTQ file). Replace the decimal number with the correct length of the trimmed read. For example, use --length-tag 'length=' to search for fields like 'length=123'." />
+            <param argument="--rename" label="Rename reads" type="text" help="This option can be used to rename both single-end and paired-end reads. " />
```
[the remark this answers is not in the document]
Comment (gallardoalba) [reply tools/cutadapt/cutadapt.xml:267]: Yes, also in length tag I think. I'll include also this option https://cutadapt.readthedocs.io/en/stable/guide.html#overview-of-adapter-types

File: tools/cutadapt/cutadapt.xml [hunk 3]
```diff
@@ -187,128 +197,178 @@ $read_mod_options.trim_n
...
+                    <param argument="--length" type="integer" value="0" label="Length" help="Shorten reads to this length. This modification is applied after adapter trimming." />
+                    <param name="shorten_end" type="select" label="End at which to apply the slice">
+                        <option value="3prime">3' ends</option>
+                        <option value="5prime">5' ends</option>
+                    </param>
+                </when>
+                <when value="False">
+                </when>
+            </conditional>
+            <param argument="--length-tag" label="Length Tag" type="text" help="Search for TAG followed by a decimal number in the name of the read (description/comment field of the FASTA or FASTQ file). Replace the decimal number with the correct length of the trimmed read. For example, use --length-tag 'length=' to search for fields like 'length=123'." />
```
Comment (bgruening) [inline tools/cutadapt/cutadapt.xml:287]: validator?

**Review by bgruening — COMMENTED** [review]
Very nice, final comment. Thanks for the bunch of tests.

File: tools/cutadapt/cutadapt.xml (hunk 3 above)
Comment (gallardoalba) [reply tools/cutadapt/cutadapt.xml]: Done! :)

**Review by bgruening — APPROVED** [review]

**Pull request merged**